pkg/lib/serverTime.js

```diff
@@ -439,63 +610,142 @@ function ValidatedInput({ errors, error_key, children }) {
         event.preventDefault();
         return false;
     }
 
-    const ntp_servers = (
+    if (servers.length == 0)
+        servers.push("");
+
+    const server_rows = (
         <table>
             <tbody>
-                { custom_ntp.servers.map((s, i) => (
+                { servers.map((s, i) => (
                     <tr key={i}>
                         <td style={{ width: "100%" }}>
                             <input type="text" className="form-control" value={s} placeholder={_("NTP server")}
                 onChange={event => change_server(event, i, event.target.value)} />
                         </td>
                         <td>
-                            <Button variant="secondary" onClick={event => add_server(event, i)} icon={ <PlusIcon /> } />
+                            <Button variant="secondary" onClick={event => add_server(event, i)}
+                                    icon={ <PlusIcon /> } className="add-server" />
                         </td>
                         <td>
                             <Button variant="secondary" onClick={event => remove_server(event, i)}
-                                    icon={ <CloseIcon /> }
-                                    isDisabled={i == custom_ntp.servers.length - 1} />
+                                    icon={ <CloseIcon /> } className="remove-server"
+                                    isDisabled={servers.length == 1 && servers[0] == ""} />
                         </td>
                     </tr>))
                 }
             </tbody>
         </table>);
 
+    return (
+        <Validated errors={errors} error_key="ntp_servers">
+            <div id="systime-ntp-servers-row">
+                <div id="systime-ntp-servers">
+                    { server_rows }
+                </div>
+            </div>
+        </Validated>);
+};
+
+const StaticServerList = ({ servers, change, map }) => {
+    function label(name) {
+        if (map && map[name])
+            return map[name];
+        else
+            return name;
+    }
+    return <>
+        { servers.map(s => <Checkbox key={s.name}
+                                            id={"cb-" + s.name}
+                                            label={label(s.name)}
+                                            isChecked={s.enabled}
+                                            onChange={val => { s.enabled = val; change() }} />)
+        }
+    </>;
+};
+
+const TimesyncdConfig = ({ state, errors, change }) => {
+    return <>
+        <Checkbox id="cb-specific"
+                         label={_("Use specific time servers")}
+                         isChecked={state.enabled}
+                         onChange={val => { state.enabled = val; change() }} />
+        { state.enabled &&
+        <DynamicServerList servers={state.servers} errors={errors} change={change} />
+        }
+    </>;
+};
+
+const chrony_sources_map = {
+    "/run/chrony-dhcp": _("NTP servers received via DHCP"),
```

Comment:
I'd try hard to avoid such hardcoded paths. If we can do something more sensible with parsing `sourcedir`s, this should just be a particular case of that. Some `/run/.*dhcp.*` matching is nice for the i18n aspect, though.